c2rust-analyze/src/context.rs

```diff
@@ -164,14 +166,46 @@ impl<'tcx> GlobalAnalysisCtxt<'tcx> {
         *next_ptr_id = counter;
     }
 
-    pub fn label_field(&mut self, field: &FieldDef) {
-        let lty = self.assign_pointer_ids(self.tcx.type_of(field.did));
+    pub fn assn_ptr_to_field(&mut self, field: &FieldDef) {
+        let mut lty = self
+            .lcx
+            .label(self.tcx.type_of(field.did), &mut |_| PointerId::NONE);
+
+        // If this is an Adt, erase any lifetime bounds because they will
+        // be erased in the MIR body when visiting.
```

Comment:
Why is this necessary?  I think we mostly ignore regions appearing in the MIR or elsewhere.

Also, `TyCtxt` has an `erase_regions` method, which you might be able to use instead of the `label` + `rewrite_unlabeled` logic here.

Reply:
the region types aren't ignored when checking the types of e.g. an assignment lhs and rhs. i'll look into the `erase_regions` method

Reply:
>the region types aren't ignored when checking the types of e.g. an assignment lhs and rhs in the assertions made.

I think those are just debug assertions.  We could presumably change the assertions to ignore regions when checking for equality, though erasing regions up front is probably easier (I couldn't find an existing "compare ignoring regions" in the rustc docs).

Reply:
addressed in 04656e0d

Reply:
Looks good.  I wasn't sure about the performance implications of calling `tcx.erase_regions` in so many places, but apparently the results of that operation are cached.